test/unit/controllers/declaration/TaricCodeControllerSpec.scala

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import base.TestHelper
+import controllers.declaration.TaricCodeController
+import controllers.util.{Add, SaveAndContinue}
+import forms.declaration.TaricCode
+import forms.declaration.TaricCode.taricCodeKey
+import models.DeclarationType.DeclarationType
+import models.{DeclarationType, Mode}
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
+import org.scalatest.OptionValues
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import unit.mock.ErrorHandlerMocks
+import views.html.declaration.taric_codes
+
+class TaricCodeControllerSpec extends ControllerSpec with ErrorHandlerMocks with OptionValues {
+
+  val mockPage = mock[taric_codes]
+
+  val controller =
+    new TaricCodeController(
+      mockAuthAction,
+      mockJourneyAction,
+      mockErrorHandler,
+      mockExportsCacheService,
+      navigator,
+      stubMessagesControllerComponents(),
+      mockPage
+    )(ec)
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    when(mockPage.apply(any(), any(), any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    reset(mockPage)
```

Comment:
We should first reset after that use `super.afterEach()`. It's just logical order, not mistake